src/main/java/com/stripe/model/Source.java

```diff
@@ -219,6 +220,25 @@ public Source detach(Map<String, Object> params, RequestOptions options) throws
         ApiResource.RequestMethod.DELETE, url, params, Source.class, options);
   }
 
+  /** Delete a specified source for a given customer. */
+  public Source detach(SourceDetachParams params) throws StripeException {
+    return detach(params, (RequestOptions) null);
+  }
+
+  /** Delete a specified source for a given customer. */
+  public Source detach(SourceDetachParams params, RequestOptions options) throws StripeException {
+    String url =
+        String.format(
+            "%s%s",
+            Stripe.getApiBase(),
+            String.format(
+                "/v1/customers/%s/sources/%s",
+                ApiResource.urlEncodeId(this.getCustomer()),
+                ApiResource.urlEncodeId(this.getId())));
+    return ApiResource.request(
+        ApiResource.RequestMethod.DELETE, url, params, Source.class, options);
+  }
```

Comment:
@ob-stripe confirming that we're okay adding that endpoint to `Source` instead of requiring they go through the Delete on the Customer instead?

Reply:
`Source.detach` already existed, we just lacked the overloads for typed parameters.